### PR TITLE
Feature slipstream

### PR DESCRIPTION
**Pull request by mclarke2**
This pull request is the first of four large pull request (Slipstream, Plots, VTOL and VLM) in that order.
Since there are significant changes in code structure please take some time to go through. These changes may not necessarily be all related to the branch but rather are related to all four branches.
The major changes are:
- Addition of a new aerodynamics method (Blown Wing Aero which does not use a surrogate, rather queries each point in the mission. NB. the propeller slipstream  and its influence changes at each point in the mission 
- Addition of propeller slipstream influence on the wing (Blown Wing VLM). 
- Removal of "prop_attributes" from propeller data structure
- Update of propeller scripts to store data for VTOL and Noise analysis
- * Removal of evtol weights methods since the vtol branch has not been pulled in. I am not sure how this passes since we have different networks for different vehicles 
- Addition of Internal Combustion Engine

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -72,6 +72,7 @@ nosetests.xml
 *.png
 
 ide/Wing/SUAVE.wpr
+/ide/Wing/SUAVE.wpr
```
Comment (WallyMaier) [inline .gitignore]: This looks unnecessary

File: regression/automatic_regression.py [hunk 2]
```diff
@@ -34,13 +34,13 @@
 modules = [
 
     # ----------------------- Regression List --------------------------
-    'scripts/weights/eVTOL_Weights_Buildup_Regression.py',
```
Comment (WallyMaier) [inline regression/automatic_regression.py]: why is this removed?

File: regression/scripts/VTOL/mission_QuadShot.py [hunk 3]
```diff
@@ -0,0 +1,550 @@
+# vtol_UAV.py
+# 
+# Created:  Jan 2016, E. Botero
+# Modified: 
+
+#----------------------------------------------------------------------
+#   Imports
```
Comment (WallyMaier) [inline regression/scripts/VTOL/mission_QuadShot.py]: i dont think is called in the automatic regression script?

File: regression/scripts/VTOL/test_VTOL.py [hunk 4]
```diff
@@ -0,0 +1,109 @@
+# test_VTOL.py
+# 
+# Created:  July 2018, M. Clarke
+#
+""" setup file for a mission with a QuadShot
```
Comment (WallyMaier) [inline regression/scripts/VTOL/test_VTOL.py]: the regression progression here is a bit confusing?  does slip stream call test_vtol?  if not, this isnt called.

File: regression/scripts/solar_network/solar_network.py [hunk 5]
```diff
@@ -61,30 +61,28 @@ def main():
...
-    prop_attributes.angular_velocity    = RPM*(2.*np.pi/60.0)
-    prop_attributes.tip_radius          = Radius
-    prop_attributes.hub_radius          = Hub_Radius
-    prop_attributes.design_Cl           = Design_Cl 
-    prop_attributes.design_altitude     = design_altitude
-    prop_attributes.design_thrust       = Thrust
-    prop_attributes.design_power        = Power
-    prop_attributes                     = propeller_design(prop_attributes)
+    prop                 = SUAVE.Components.Energy.Converters.Propeller()
+    prop.number_blades       = Blades
```
Comment (WallyMaier) [inline regression/scripts/solar_network/solar_network.py]: is this just a refactoring?  why prop_attributes -> prop?   (just curious)

File: regression/scripts/weights/eVTOL_Weights_Buildup_Regression.py [hunk 6]
```diff
@@ -1,364 +0,0 @@
-# eVTOL_Weights_Buildup_Regression.py
-#
-# Created: Feb, 2018, J. Smart
-
-#------------------------------------------------------------------------------
```
Comment (WallyMaier) [inline regression/scripts/weights/eVTOL_Weights_Buildup_Regression.py]: why gone?

File: trunk/SUAVE/.idea/SUAVE.iml [hunk 7]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (WallyMaier) [inline trunk/SUAVE/.idea/SUAVE.iml]: are these supposed to be here?

File: trunk/SUAVE/.idea/misc.xml [hunk 8]
```diff
@@ -0,0 +1,4 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (WallyMaier) [inline trunk/SUAVE/.idea/misc.xml]: is this necessary?

File: trunk/SUAVE/.idea/modules.xml [hunk 9]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (WallyMaier) [inline trunk/SUAVE/.idea/modules.xml]: is this necessary?

File: trunk/SUAVE/.idea/workspace.xml [hunk 10]
```diff
@@ -0,0 +1,100 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
```
Comment (WallyMaier) [inline trunk/SUAVE/.idea/workspace.xml]: is this necessary?

File: trunk/SUAVE/Components/Energy/Converters/Internal_Combustion_Engine.py [hunk 11]
```diff
@@ -0,0 +1,149 @@
+# Internal_Combustion_Engine.py
```
Comment (WallyMaier) [inline trunk/SUAVE/Components/Energy/Converters/Internal_Combustion_Engine.py]: this is new?  is it used?

File: trunk/SUAVE/Components/Energy/Converters/Motor_Lo_Fid.py [hunk 12]
```diff
@@ -157,4 +159,26 @@ def current(self,conditions):
...
+        i = power_out/(etam*v_in)
+        
+        i[power_out == 0.] = 0.
+        
+        self.outputs.power   = power_out
+        self.outputs.current = i
+        
+        return power_out, i
+    
+
```
Comment (WallyMaier) [inline trunk/SUAVE/Components/Energy/Converters/Motor_Lo_Fid.py]: white space

File: trunk/SUAVE/Components/Energy/Networks/Internal_Combustion_Propeller.py [hunk 13]
```diff
@@ -0,0 +1,153 @@
...
+# package imports
+import numpy as np
+from SUAVE.Components.Propulsors.Propulsor import Propulsor
+from SUAVE.Core import Data, Units
+
+# ----------------------------------------------------------------------
+#  Network
+# ----------------------------------------------------------------------
+## @ingroup Components-Energy-Networks
+class Internal_Combustion_Propeller(Propulsor):
```
Comment (WallyMaier) [inline trunk/SUAVE/Components/Energy/Networks/Internal_Combustion_Propeller.py]: not quite sure what this entails?  what makes this different than a regular prop?  maybe a more clear description or name?

File: trunk/SUAVE/Components/Energy/Networks/Stopped_Rotor.py [hunk 14]
```diff
@@ -129,23 +129,23 @@ def evaluate_thrust(self,state):
...
         
         # ESC Voltage
         esc_lift.inputs.voltagein    = volts      
         esc_forward.inputs.voltagein = volts 
         
-        ###
-        # Evaluate thrust from the forward propulsors
-        ###
+        ##
+        # EVALUATE THRUST FROM FORWARD PROPULSORS
```
Comment (WallyMaier) [inline trunk/SUAVE/Components/Energy/Networks/Stopped_Rotor.py]: i like this convention....all uppercase.  We should be aware to make this change in future pull 
(imo)

File: trunk/SUAVE/Input_Output/OpenVSP/vsp_write.py [hunk 15]
```diff
@@ -691,7 +689,7 @@ def write_vsp_fuselage(fuselage,area_tags, main_wing, fuel_tank_set_ind):
         if 'z_pos' in vals.tail:
             tail_z_pos = vals.tail.z_pos
         else:
-            pass # use above default        
+            pass # use above default
```
Comment (WallyMaier) [inline trunk/SUAVE/Input_Output/OpenVSP/vsp_write.py]: remove this space

File: trunk/SUAVE/Methods/Missions/Segments/Common/Aerodynamics.py [hunk 16]
```diff
@@ -197,6 +197,7 @@ def update_aerodynamics(segment):
     L = segment.state.ones_row(3) * 0.0
     D = segment.state.ones_row(3) * 0.0
 
+
```
Comment (WallyMaier) [inline trunk/SUAVE/Methods/Missions/Segments/Common/Aerodynamics.py]: white space

File: trunk/SUAVE/Methods/Missions/Segments/Hover/Climb.py [hunk 17]
```diff
@@ -47,7 +49,7 @@ def initialize_conditions(segment):
     if alt0 is None:
         if not segment.state.initials: raise AttributeError('initial altitude not set')
         alt0 = -1.0 * segment.state.initials.conditions.frames.inertial.position_vector[-1,2]
-
+
```
Comment (WallyMaier) [inline trunk/SUAVE/Methods/Missions/Segments/Hover/Climb.py]: remove the spaces/tab

File: trunk/SUAVE/Methods/Noise/Fidelity_One/Noise_Tools/noise_geometric.py [hunk 18]
```diff
@@ -150,4 +150,57 @@ def noise_geometric(noise_segment,analyses,config):
     noise_segment.theta = theta
     noise_segment.phi   = phi
 
-    return (dist,theta,phi)
\ No newline at end of file
+    return (dist,theta,phi)
+
+
```
Comment (WallyMaier) [inline trunk/SUAVE/Methods/Noise/Fidelity_One/Noise_Tools/noise_geometric.py]: white space

File: trunk/SUAVE/Methods/Noise/Fidelity_One/Noise_Tools/noise_geometric.py [hunk 19]
```diff
@@ -150,4 +150,57 @@ def noise_geometric(noise_segment,analyses,config):
...
+    #--------------------------------------------------------
+    #------------------ FLYOVER CALCULATION -----------------
+    #--------------------------------------------------------
+    
+    # Azimuthal angle is zero for flyover condition
+    phi=np.zeros(n_steps)    
+    theta = np.zeros(n_steps)  
+       
+    
+  #  S_0 = 1500 * Units.ft
```
Comment (WallyMaier) [inline trunk/SUAVE/Methods/Noise/Fidelity_One/Noise_Tools/noise_geometric.py]: what is this?  also, white space above

File: trunk/SUAVE/Methods/Weights/Buildups/Electric_Stopped_Rotor/empty.py [hunk 20]
```diff
@@ -75,14 +75,15 @@ def empty(config,
 # Unpack Inputs
 #-------------------------------------------------------------------------------
 
-    rProp               = config.propulsors.network.propeller.prop_attributes.tip_radius
-    mBattery            = config.propulsors.network.battery.mass_properties.mass
-    mPayload            = config.propulsors.network.payload.mass_properties.mass
+    rPropLift           = config.propulsors.propulsor.propeller_lift.tip_radius
```
Comment (WallyMaier) [inline trunk/SUAVE/Methods/Weights/Buildups/Electric_Stopped_Rotor/empty.py]: is this a suave-wide change?  network -> propulsors?

File: trunk/SUAVE/Optimization/Nexus.py [hunk 21]
```diff
@@ -1,4 +1,4 @@
-## @ingroup Optimization
+ ## @ingroup Optimization
```
Comment (WallyMaier) [inline trunk/SUAVE/Optimization/Nexus.py]: remove space

Comment (mclarke2) [reply trunk/SUAVE/Optimization/Nexus.py]: I'm going to correct this and delete this pull request per our discussion this morning

File: trunk/SUAVE/Methods/Weights/Buildups/Electric_Stopped_Rotor/empty.py (hunk 20 above)
Comment (mclarke2) [reply trunk/SUAVE/Methods/Weights/Buildups/Electric_Stopped_Rotor/empty.py]: It is. I'm going to correct this and delete this pull request per our discussion this morning

File: trunk/SUAVE/Methods/Noise/Fidelity_One/Noise_Tools/noise_geometric.py (hunk 19 above)
Comment (mclarke2) [reply trunk/SUAVE/Methods/Noise/Fidelity_One/Noise_Tools/noise_geometric.py]: I'm going to correct this and delete this pull request per our discussion this morning

File: trunk/SUAVE/Methods/Noise/Fidelity_One/Noise_Tools/noise_geometric.py (hunk 18 above)
Comment (mclarke2) [reply trunk/SUAVE/Methods/Noise/Fidelity_One/Noise_Tools/noise_geometric.py]: I'm going to correct this and delete this pull request per our discussion this morning

File: trunk/SUAVE/Methods/Missions/Segments/Hover/Climb.py (hunk 17 above)
Comment (mclarke2) [reply trunk/SUAVE/Methods/Missions/Segments/Hover/Climb.py]: I'm going to correct this and delete this pull request per our discussion this morning

File: regression/scripts/solar_network/solar_network.py (hunk 5 above)
Comment (mclarke2) [reply regression/scripts/solar_network/solar_network.py]: Yes, no more prop attributes. I'm going to correct this and delete this pull request per our discussion this morning

File: regression/automatic_regression.py (hunk 2 above)
Comment (mclarke2) [reply regression/automatic_regression.py]: This is temporarily removed. Will be included in the VTOL branch

File: .gitignore (hunk 1 above)
Comment (mclarke2) [reply .gitignore]: Will remove

File: regression/scripts/VTOL/mission_QuadShot.py (hunk 3 above)
Comment (mclarke2) [reply regression/scripts/VTOL/mission_QuadShot.py]: This is temporarily removed. Will be included in the VTOL branch.

File: regression/automatic_regression.py (hunk 2 above)
Comment (mclarke2) [reply regression/automatic_regression.py]: The Weights Branch will pulled in prematurely without the VTOL networks. As a result the quickfix to create the old pull request do not work (without the VTOL branch)

File: regression/scripts/VTOL/mission_QuadShot.py (hunk 3 above)
Comment (mclarke2) [reply regression/scripts/VTOL/mission_QuadShot.py]: The Weights Branch will pulled in prematurely without the VTOL networks. As a result the quickfix to create the old pull request do not work (without the VTOL branch)
